ckanext/archiver/plugin.py

```diff
@@ -205,6 +210,16 @@ def after_show(self, context, pkg_dict):
                 del archival_dict['resource_id']
                 res['archiver'] = archival_dict
 
+    # IRoutes
+
+    def before_map(self, map):
```

Comment:
What about the existing `resource_cache` route?

Reply:
archiver states you need to update nginx/apache with mapping the filesystem to be shown to the public. If will still need to fall back on that somehow for on disk files (so this update would be accepted by the maintainers). 

reality is that it should use the uploader function with a subfolder difference and be able to allow downloads. 
It could be still all inside upload but with an extra folder so resource download could still work.